framework/tests/whole-program-inference/non-annotated/OverriddenMethodsTest.java

```diff
@@ -0,0 +1,28 @@
+import tests.wholeprograminference.qual.*;
+class Parent {
+    public void foo(@Sibling1 Object obj, @Sibling2 Object obj2) {}
+    public void bar(@Sibling1 Parent this, @Sibling2 Object obj) {}
+    public void barz(@Sibling1 Parent this, @Sibling2 Object obj) {}
+}
+
+class Child extends Parent {
+    @Override
+    public void foo(Object obj, Object obj2) {
+        //:: error: (assignment.type.incompatible)
+        @Sibling1 Object o = obj;
+        //:: error: (assignment.type.incompatible)
+        @Sibling2 Object o2 = obj2;
+    }
+
+    @Override
+    public void bar(Object obj) {
+        //:: error: (assignment.type.incompatible)
+        @Sibling1 Child child = this;
+        //:: error: (assignment.type.incompatible)
+        @Sibling2 Object o = obj;
+    }
+
+    @SuppressWarnings("")
+    @Override
+    public void barz(Object obj) {}
```

Comment:
I manually verified that @Sibling2 wasn't added to parameter of Child.barz after the inference process, but I couldn't find a way to automatically test that. I removed this SuppressWarnings test.